### PR TITLE
[FE-8233] Improve error handling in getFields

### LINKED ISSUE #15: Handle undefined `row_desc` in `getFields` Method
In the `getField` method, we create a row dictionary in a way that can cause a runtime error when the return `fields` object has an undefined `row_desc` key

```js
const rowDict = fields.row_desc.reduce((accum, value) => {
  accum[value.col_name] = value
    return accum
}, {})
```

### PR DESCRIPTION
**Pull request by jrajav**
Improve error handling for getFields call to cover case of unknown error from callback, as well as row_desc being undefined (shouldn't happen).

- Fixes #15 
- Fixes [FE-8233](https://jira.omnisci.com/browse/FE-8233)

### CODE REVIEW

File: src/mapd-con-es6.js [hunk 1]
```diff
@@ -1128,12 +1128,18 @@ class MapdCon {
       this._sessionId[0],
       tableName,
       (error, fields) => {
-        if (fields) {
-          const rowDict = fields.row_desc.reduce((accum, value) => {
-            accum[value.col_name] = value
-            return accum
-          }, {})
-          callback(null, this.convertFromThriftTypes(rowDict))
+        if (error) {
```
Comment (israelvicars) [inline src/mapd-con-es6.js]: @jrajav Will the `else` ever get hit with this condition in place?

Comment (israelvicars) [reply src/mapd-con-es6.js]: Referring to the outer `else` on ln 1143

Comment (israelvicars) [reply src/mapd-con-es6.js]: Ok. If there is no error for that case, then ` + error` could be removed from ln 1144.

Comment (jrajav) [reply src/mapd-con-es6.js]: Good call, thanks! Fixed with latest commit

**Comment by jrajav** [comment]
Tested https://github.com/omnisci/mapd-connector/pull/138/commits/dfb78cf6b08bb8b3478cbe4e6551565f0fcdc130 manually against latest Core.

**Pull request merged**